OpenSim/Tools/CMCTool.cpp

```diff
@@ -433,7 +434,34 @@ bool CMCTool::run()
     _model->getMultibodySystem().realize(s, Stage::Position );
      taskSet.setModel(*_model);
     _model->equilibrateMuscles(s);
-  
+    
+    SimTK::GeneralForceSubsystem& gfs = _model->updForceSubsystem();
+    int autoThreadCount = gfs.getNumberOfThreads(); //auto-detected numThreads
+    char* envThreadsString = getenv("OPENSIM_CMC_MAX_THREADS");
+    //TODO: Best environment variable name?
+    int envThreads = autoThreadCount;
+    
+    cout << endl;
+    if(envThreadsString != NULL)
+    {
+      envThreads = stoi(envThreadsString);
+      if(autoThreadCount != envThreads)
+      {
+          gfs.setNumberOfThreads(envThreads);
+          cout << "Enviroment variable OPENSIM_CMC_MAX_THREADS set to " <<
+          envThreads << ". Overriding the detected number of " <<
+          autoThreadCount << " threads." << endl;
+      }else{
+        cout << "Enviroment variable OPENSIM_CMC_MAX_THREADS set to " <<
+        envThreads << ". Same as the detected number of " <<
+        autoThreadCount << " threads." << endl;
+      }
+
+    }else{
+      cout << "Environment variable OPENSIM_CMC_MAX_THREADS not set! Using the"
+      " detected number of " << autoThreadCount << " threads." << endl;
+    }
+
```

Comment:
I just ran CMC; this works great :). However I wonder if we could do better error handling. If I set `OPENSIM_CMC_MAX_THREADS` to 0, I get this exception:

```
64: SimTK Exception thrown at ParallelExecutor.cpp:51:
64:   Bad call to Simbody API method ParallelExecutorImpl::ParallelExecutorImpl(): Number of threads must be positive.
64:   (Required condition 'numThreads > 0' was not met.)
```

Do you think we could give OpenSim users a better exception message that connects the error to the environment variable? Either do a separate check in CMCTool or catch the above exception and throw a new OpenSim exception based on it.